src/test/clojure/xtdb/tutorial_test.clj

```diff
@@ -14,69 +14,133 @@
       x-json/parse-query
       x-edn/unparse))
 
+(def users
+  [[:put :users {:xt/id :ivan, :first-name "Ivan", :last-name "Ivanov", :age 25}]
+   [:put :users {:xt/id :petr, :first-name "Petr", :last-name "Petrov", :age 27}]])
+
+(def old-users
+  (mapv #(update-in % [2 :age] + 30) users))
+
+(def authors
+  [[:put :authors {:xt/id :ivan, :first-name "Ivan", :last-name "Ivanov"}]
+   [:put :authors {:xt/id :petr, :first-name "Petr", :last-name "Petrov"}]])
+
+(def articles
+  [[:put :articles {:xt/id 1, :author-id :ivan, :title "First" :content "My first blog"}]
+   [:put :articles {:xt/id 2, :author-id :ivan, :title "Second" :content "My second blog"}]])
+
+(def posts
+  [[:put :posts {:xt/id 1, :author-id :ivan, :title "First" :content "My first blog"}]
+   [:put :posts {:xt/id 2, :author-id :ivan, :title "Second" :content "My second blog"}]])
+
+(def comments
+  [[:put :comments {:xt/id 1, :article-id 1, :post-id 1, :created-at #inst "2020", :comment "1"}]
+   [:put :comments {:xt/id 2, :article-id 1, :post-id 1, :created-at #inst "2021", :comment "2"}]
+   [:put :comments {:xt/id 3, :article-id 2, :post-id 2, :created-at #inst "2022", :comment "1"}]])
+
+(def customers
+  [[:put :customers {:xt/id :ivan}]
+   [:put :customers {:xt/id :petr}]])
+
+(def orders
+  [[:put :orders {:xt/id 1, :customer-id :ivan, :currency :gbp, :order-value 100}]
+   [:put :orders {:xt/id 2, :customer-id :ivan, :currency :gbp, :order-value 150}]])
+
+(def promotions
+  [[:put :promotions {:xt/id 1, :promotion-type :christmas}]
+   [:put :promotions {:xt/id 2, :promotion-type :easter}]])
+
+(def documents
+  [[:put :documents {:xt/id "doc-id", :version 1}]
+   [:put :documents {:xt/id "doc-id2", :version 1}]])
+
```

Comment:
If you only use data once then inline it in the submit call directly. Otherwise this is good.